### PR TITLE
Cherry pick commits from main

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This Pr is to cherry pick the commits from `main` branch to `release-v2.8` branch since `v2.8.0-beta.3` for the prerelease of `v2.8.0-rc.1`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,18 @@
 # Changes to the Mapbox Navigation SDK for iOS
 
+## v2.9.0
+
+### Map
+
+* `NavigationMapView.removeAlternativeRoutes()` and `NavigationMapView.removeContinuousAlternativeRoutesDurations()` were made public to provide a way to remove previously shown alternative routes and alternative routes duration annotations, respectively. ([#4134](https://github.com/mapbox/mapbox-navigation-ios/pull/4134))
+
```
Comment (MaximAlien) [inline CHANGELOG.md]: Looks like this should not be a part of `v2.8.0`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,7 +52,11 @@
 * You can now set `Waypoint.allowsSnappingToStaticallyClosedRoad` to `true` to allow the waypoint to snap to a road that is fully closed for long-term construction. ([#4085](https://github.com/mapbox/mapbox-navigation-ios/pull/4085))
 * Added `NavigationSettings.liveIncidentsOptions` to configure how incidents data is fetched. ([#4088](https://github.com/mapbox/mapbox-navigation-ios/pull/4088))
 * Added `NavigationSettings.statusPollingConfig` to configure navigator status polling options. ([#4135](https://github.com/mapbox/mapbox-navigation-ios/pull/4135))
+* Added `RouterDelegate.router(_:modifiedOptionsForReroute:)`, `NavigationServiceDelegate.navigationService(_:modifiedOptionsForReroute:)` and `NavigationViewControllerDelegate.navigationViewController(_:modifiedOptionsForReroute:)` methods to allow `RouteOptions` customization on reroutes. ([#4102](https://github.com/mapbox/mapbox-navigation-ios/pull/4102))
```
Comment (MaximAlien) [inline CHANGELOG.md:50]: @Udumft, do you think that it should be part of `v2.8.0`? IMHO it'd be safer to land it in `v2.9.0` and remove from cherry-picked commits.

File: CHANGELOG.md (hunk 1 above)
Comment (ShanMa1991) [reply CHANGELOG.md]: Thanks for the suggestion, this part has been updated and moved to v2.8.0

**Review by MaximAlien — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Udumft) [reply CHANGELOG.md:49]: It was not required to be picked to 2.8...

**Pull request merged**